gateway/src/metrics.rs

```diff
@@ -782,8 +782,9 @@ impl SpPoller {
                     //
                     // See this comment for details on what Tctl values mean:
                     // https://github.com/illumos/illumos-gate/blob/6cf3cc9d1e40f89e90135a48f74f03f879fce639/usr/src/uts/intel/io/amdzen/smntemp.c#L21-L57
-                    let is_tctl =
-                        sensor == "CPU" && target.component_kind == "sbtsi";
+                    let is_tctl = (sensor == "CPU"
+                        && target.component_kind == "sbtsi")
+                        || m.kind == MeasurementKind::CpuTctl;
```

Comment:
@hawkw Is this change right? I'm wondering if we need to continue supporting "temperator + CPU + sbtsi is really cpu tctl" now that we have an explicit enum variant for it (and would like a double-check that this new variant should be treated this way!). Maybe it depends on what SP versions we assume we're talking to?